### PR TITLE
Add local Git repository creation

### PR DESCRIPTION
**Pull request by davix93**
# Description

This PR adds the possibility to create a git local repository when selecting a workspace, if a `.git` folder does not already exist.

Implements #77 

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Manually tested

**Configuration**:
OS and Version: macOS 11.2.3

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

# Screenshots

| Add git repo selected | Add git repo unavailable |
|------|------|
|||

### CODE REVIEW

File: Sources/App/Views/Sections/Settings/Server/ServerSettingsViewModel.swift [hunk 1]
```diff
@@ -117,6 +127,10 @@ final class ServerSettingsViewModel: ObservableObject {
         ServerConnectionConfiguration(hostname: hostname, port: Int(port) ?? 8080)
       )
 
+      if shouldCreateGitRepository {
```
Comment (FabrizioBrancati) [inline Sources/App/Views/Sections/Settings/Server/ServerSettingsViewModel.swift:130]: Why not use `isGitRepositoryCreationEnabled`? We will avoid the `shouldCreateGitRepository` parameter

**Review by FabrizioBrancati — CHANGES_REQUESTED** [review]

File: Sources/App/Views/Sections/Settings/Server/ServerSettingsViewModel.swift (hunk 1 above)
Comment (TheInkedEngineer) [reply Sources/App/Views/Sections/Settings/Server/ServerSettingsViewModel.swift:130]: 👍🏼

**Review by TheInkedEngineer — APPROVED** [review]
I agree with what @FabrizioBrancati said. Once that is fixed, you're good to go.

File: Sources/App/Views/Sections/Settings/Server/ServerSettingsViewModel.swift (hunk 1 above)
Comment (davix93) [reply Sources/App/Views/Sections/Settings/Server/ServerSettingsViewModel.swift]: done 👍

**Review by FabrizioBrancati — APPROVED** [review]
LGTM 👍

**Pull request merged**